addons/payment/tests/test_res_company.py

```diff
@@ -0,0 +1,22 @@
+from odoo.tests import tagged
+
+from odoo.addons.payment.tests.common import PaymentCommon
+
+
+@tagged('-at_install', 'post_install')
+class TestResCompany(PaymentCommon):
+
+    def test_creating_company_duplicates_providers(self):
+        """Ensure that payment providers of an existing company are correctly duplicated
+        when a new company is created."""
+        main_company = self.env.company
+        main_company_providers = self.env['payment.provider'].search_count(
```

Comment:
The variable ends with "providers", so it should either use `search`, or end with "count".